### PR TITLE
Handle calculate checkout total zero and negative

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because we want to skip running tax plugin if checkout total is zero. This way we also prevent total net from being negative by subtracting full discount gross amount during tax calculations.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5696 totalPrice.amount.net takes a negative value](https://app.clickup.com/t/2549495/SALEOR-5696)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/checkout/base_calculations.py [hunk 1]
```diff
@@ -182,12 +182,15 @@ def calculate_base_price_for_shipping_method(
...
     shipping_price: TaxedMoney,
-    discount: TaxedMoney,
+    discount: Money,
     currency: str,
 ) -> TaxedMoney:
     """Return the total cost of the checkout."""
+    zero = zero_taxed_money(currency)
     total = subtotal + shipping_price - discount
-    return max(total, zero_taxed_money(currency))
+    if total.gross <= zero.gross:
```
Comment (korycins) [inline saleor/checkout/base_calculations.py:191]: Can we add an explanation here? For someone not familiar with the issue, this could not be clear.

**Review by maarcingebala — APPROVED** [review]

File: saleor/checkout/base_calculations.py [hunk 2]
```diff
@@ -182,12 +182,19 @@ def calculate_base_price_for_shipping_method(
...
     """Return the total cost of the checkout."""
+    zero = zero_taxed_money(currency)
     total = subtotal + shipping_price - discount
-    return max(total, zero_taxed_money(currency))
+    # Discount is subtracted from both gross and net values, which may cause negative
+    # net value if we are having a discount that covers whole price.
+    # Comparing TaxedMoney objects works only on gross values. That is why we are
+    # explicitly returning zero_taxed_money if total.gross is less or equal zero.
+    if total.gross <= zero.gross:
+        return zero_taxed_money(currency)
```
Comment (L3str4nge) [inline saleor/checkout/base_calculations.py:196]: We can just return `zero` instead of calling this function again

**Review by L3str4nge — APPROVED** [review]

File: saleor/checkout/base_calculations.py (hunk 1 above)
Comment (jakubkuc) [reply saleor/checkout/base_calculations.py:195]: added.

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**